### ISSUE TITLE
[DOC-280] Single sign-on (SSO)

### ISSUE THREAD

**Issue by ElTimuro** [documenso/backlog#16]
* SSO ([https://en.wikipedia.org/wiki/Single_sign-on](https://en.wikipedia.org/wiki/Single_sign-on)) can improve and simplify the user's experience
* Offering a solid SSO is an enabler for more enterprise usage.

<sub>From [SyncLinear.com](https://synclinear.com) | [DOC-280](https://linear.app/documenso/issue/DOC-280/single-sign-on-sso)</sub>

**Comment by aaronfisher** [comment]
SSO (alongside Salesforce) is necessary for any sort of Enterprise adoption, to kick DocuSign out of the flow. Looking forward to the day.

**Comment by ElTimuro** [comment]
@dguyen @Mythie this is related: https://github.com/documenso/backlog-internal/issues/317

from what i can tell, user sync is more requested. Though I assume they go together anyway?

**Comment by dguyen** [comment]
@ElTimuro Need to do this first then will do user sync after for architectural purposes

TLDR of the feature

- Allow Organisations to configure an "OIDC" (such as Google SSO, Auth0) login portal to allow members to use to sign in.
- Users who sign in through this portal will automatically join the organisation
